lib/active_model/serializer/adapter/json.rb

```diff
@@ -6,45 +6,128 @@ class Adapter
       class Json < Adapter
         def serializable_hash(options = nil)
           options ||= {}
+          serialized_hash = {}
+
           if serializer.respond_to?(:each)
-            @result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            serialized_hash = serialize_array_without_root(serializer, options)
           else
-            @hash = {}
+            serialized_hash = resource_object_for(serializer)
+            serialized_hash = add_resource_relationships(serialized_hash, serializer)
+          end
+
+          # TODO: remove usage of result instance variable from other flows / tests
+          @result = serialized_hash
```

Comment:
I think a super class or something -- tests unrelated to my stuff fail if I get rid of it.